pkgs/by-name/fa/fastddsgen/package.nix

```diff
@@ -3,36 +3,59 @@
   stdenv,
   makeWrapper,
   fetchFromGitHub,
-  gradle_7,
-  openjdk17,
+  fetchpatch2,
+  gradle,
+  openjdk,
+  testers,
 }:
 
-let
+stdenv.mkDerivation (finalAttrs: {
   pname = "fastddsgen";
   version = "4.2.0";
 
-  gradle = gradle_7;
-
-in
-stdenv.mkDerivation {
-  inherit pname version;
-
   src = fetchFromGitHub {
     owner = "eProsima";
     repo = "Fast-DDS-Gen";
-    tag = "v${version}";
+    tag = "v${finalAttrs.version}";
     fetchSubmodules = true;
     hash = "sha256-weGS340MvPitWMgWx1cWTgYgGcQfJSTUus8EcBob7hY=";
   };
 
+  patches = [
+    # Note: PR is not yet merged
+    (fetchpatch2 {
+      url = "https://github.com/eProsima/IDL-Parser/pull/179.diff";
+      stripLen = 1;
+      extraPrefix = "thirdparty/idl-parser/";
+      includes = [ "thirdparty/idl-parser/build.gradle" ];
+      hash = "sha256-OzywQ02yaMnya+536DeHWeKwZefI4meYqmZcp3onwR8=";
+    })
+
+    # Note: PR is not yet merged
+    (fetchpatch2 {
+      url = "https://github.com/eProsima/Fast-DDS-Gen/pull/493.diff";
+      includes = [ "build.gradle" ];
+      # TODO remove when upstream rebased the PR
+      decode = "sed 's/4\\.1\\.0/${finalAttrs.version}/g'";
```

Comment:
Using `substituteStream --replace-fail 4.1.0 ${finalAttrs.version}` is probably better because we will notice immediately when upstream rebases the PR.